readthedocs/proxito/views/serve.py

```diff
@@ -95,17 +95,25 @@ def get(self,
         storage_path = final_project.get_storage_path(
             type_='html', version_slug=version_slug, include_file=False
         )
+
+        # URL without scheme and domain to perform an NGINX internal redirect
+        storage = get_storage_class(settings.RTD_BUILD_MEDIA_STORAGE)()
+
+        # If ``filename`` is ``''`` it leaves a trailing slash
         path = os.path.join(storage_path, filename)
 
+        url = storage.url(path)  # this will remove the trailing slash
```

Comment:
I change the raw usage of `path` generated by `os.path.join` for the `storage.url` because it will return us the proper URL on the storage (the same URL that `FooterHTML` view will generate on the footer)

Also, using `storage.url` here to generate the URL will allow us to remove the code we override here in corporate (because of the `AccessKey`) and just use the same code as in community.

Reply:
This may be happening because of this https://github.com/readthedocs/readthedocs.org/blob/c2609f535981e871ff7fe979188cbbe141dd8bc3/readthedocs/builds/storage.py#L26-L36

We may want to test and refactor this if possible in another PR.